core/src/sumeragi/view_change.rs

```diff
@@ -16,90 +16,50 @@ use crate::block::{EmptyChainHash, VersionedCommittedBlock, VersionedValidBlock}
 /// The proof of a view change. It needs to be signed by f+1 peers for proof to be valid and view change to happen.
 #[derive(Debug, Clone, Decode, Encode, IntoSchema)]
 pub struct Proof {
-    payload: ProofPayload,
-    signatures: SignaturesOf<Self>,
+    pub latest_block_hash: HashOf<VersionedCommittedBlock>,
+    pub view_change_index: u64,
+    pub signatures: Vec<Signature>,
 }
 
 impl Proof {
-    /// Hash of this proof.
-    pub fn hash(&self) -> HashOf<Self> {
-        HashOf::new(&self.payload).transmute()
-    }
-
-    fn from_payload(payload: ProofPayload, key_pair: KeyPair) -> Result<Self> {
-        let signatures = SignaturesOf::new(key_pair, &payload)
-            .wrap_err("Failed to create proof of view change")?
-            .transmute();
-        Ok(Self {
-            payload,
-            signatures,
-        })
-    }
-
-    /// Constructor for `CommitTimeout` view change suggestion.
-    /// # Errors
-    /// Fails if signing failed.
-    pub fn commit_timeout(
-        hash: HashOf<VersionedValidBlock>,
-        previous_proof: HashOf<Self>,
-        latest_block: HashOf<VersionedCommittedBlock>,
-        key_pair: KeyPair,
-    ) -> Result<Self> {
-        let payload = ProofPayload {
-            reason: Reason::CommitTimeout(CommitTimeout { hash }),
-            previous_proof,
-            latest_block,
-        };
-        Self::from_payload(payload, key_pair)
-    }
+    fn signature_payload(&self) -> Hash {
+        use byteorder::{ByteOrder, LittleEndian};
```

Comment:
Please avoid having `use` statements inside functions. If the thing that you're using is local, you can rely on explicit namespacing for the thing, and if it's used in more than one function, it should be an import at the top.